src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java

```diff
@@ -259,7 +260,11 @@ String computeCrc32c(String input) {
                       + inputs.get(i)
                       + " message: "
                       + ex.getMessage());
-            } else throw new FileSystemExecutionException("batch operation failed", ex);
+            } else
+              throw new FileSystemExecutionException(
+                  "[batchOperation] Parent exception caught but neither parent nor nested "
+                      + "exceptions were designated for retry.",
+                  ex);
```

Comment:
It's weird that our tools didn't flag this, the `else` clause should be in a block
```suggestion
            } else {
              throw new FileSystemExecutionException(
                  "[batchOperation] Parent exception caught but neither parent nor nested "
                      + "exceptions were designated for retry.",
                  ex);
            }
```

Reply:
Regarding adding `batchOperation` to the logs, presumably this is because our top-level catch isn't dumping the stack crawl, only the exception? If the stack crawl is dumped, it will include the method name, which already includes `batchOperation`. Maybe the top level catch should be logging the full stack trace?

Reply:
I was just adding the [batchOperation] tag to be consistent with the rest of the method. I don't have a strong opinion about leaving or removing these.

The main change here is to be more specific about how we got to this exception. The "batch operation failed" message was not descriptive enough.